pkg/mcp/server.go

```diff
@@ -3,12 +3,34 @@ package mcp
 import (
 	"database/sql"
 	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
 
 	"github.com/mark3labs/mcp-go/server"
 	mnemonicpkg "github.com/unowned-ai/mnemonic/pkg"
 	pkgdb "github.com/unowned-ai/mnemonic/pkg/db"
 )
 
+// GetDefaultDBPath returns a system-appropriate default path for the database.
+func GetDefaultDBPath() string {
```

Comment:
This is ok for now, but it feels fragile to me long-term.

There is a third party library that correctly resolves user directories based on various cues on different operating systems: https://github.com/apparentlymart/go-userdirs

That repo looks a bit old.

There is a Python package called [`platformdirs`](https://github.com/tox-dev/platformdirs) which is more actively maintained that we can adopt the logic from. If we do this, please cite them.

Reply:
Move to separate issue
https://github.com/unowned-ai/recall/issues/38